src/zmq/zmqrpc.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <zmq/zmqrpc.h>
+
+#include <rpc/server.h>
+#include <zmq/zmqabstractnotifier.h>
+#include <zmq/zmqnotificationinterface.h>
+
+#include <univalue.h>
+
+namespace {
+
+UniValue getzmqnotifications(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 0)
+        throw std::runtime_error(
+            "getzmqnotifications\n"
+            "\nReturns information about the active ZeroMQ notifications.\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"publish\": [               (json array) Active publishers\n"
+            "    {                        (json object)\n"
+            "      \"type\": \"pubhashtx\",   (string) Type of notification\n"
+            "      \"address\": \"...\"       (string) Address of the publisher\n"
+            "    },\n"
+            "    ...\n"
+            "  ]\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("getzmqnotifications", "")
+            + HelpExampleRpc("getzmqnotifications", "")
+        );
+
+    UniValue publish(UniValue::VARR);
+    if (g_zmq_notification_interface != nullptr) {
```

Comment:
An alternative is to make these RPC available (registered) if `g_zmq_notification_interface != nullptr` so this check could be removed.

Not sure what others think.

Reply:
I was thinking about this, and my opinion is that we should have them always available (if ZMQ is compiled in at least) - as in the current code.  The reason is that if a wallet frontend or such uses this RPC, then checking whether *any* notification is registered at all (which is equivalent to checking for non-nullness of `g_zmq_notification_interface`) is part of the reason for having this RPC in the first place.  And while it may then catch the "method not found" error, this makes things more complicated.  Having the RPC and returning an empty response seems more useful for this case to me.

But of course I'm happy to change that if others disagree.

Reply:
Consider an hypothetical `zmqaddnotification`, shouldn't it raise an error if `g_zmq_notification_interface == nullptr`?

> this makes things more complicated

Why? Also, if your "frontend" is calling these RPC then it would be bad configuration if these are not present.

Reply:
But `g_zmq_notification_interface` being null just means that we have no notifications enabled through the command-line.  So with `zmqaddnotification`, I would imagine that it actually is able to add a notification also in that case (in fact, that could be the primary usage - do not configure any notifications through the CLI and instead add them later by RPC).  I. e., `zmqaddnotification` would either initialise `g_zmq_notification_interface` if it was null, or we would always initialise it (even if no notifications are specified) on startup.

(We then *might* want to have a separate argument to turn off ZMQ completely - but that would be something new and is not what we have today!)

Of course, a frontend would have to handle the case of the method failing just in case.  But I think if the daemon has ZMQ support but was just started without notifications, then the method should simply return that information (empty array), rather than not be present.  Especially with a hypothetical `addzmqnotification` in the future, this behaviour makes more sense for me.

Reply:
> But `g_zmq_notification_interface` being null just means that we have no notifications enabled through the command-line.

Indeed, forgot about that.